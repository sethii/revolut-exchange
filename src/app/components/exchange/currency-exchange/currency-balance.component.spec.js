import React from 'react';
import renderer from 'react-test-renderer';
import { toClass } from 'recompose';

import { CurrencyBalance } from './currency-balance.component';

describe('Currency balance component', () => {
  it('renders currency balance', () => {
    const CurrencyBalanceComponent = toClass(CurrencyBalance);

    const wallet = {
      code: 'USD',
      amount: 4567
    };

    expect(renderer.create(<CurrencyBalanceComponent noBalance={true} wallet={wallet}/>).toJSON()).toMatchSnapshot();
  });

  it('renders currency balance when not enough money available', () => {
    const CurrencyBalanceComponent = toClass(CurrencyBalance);

    const wallet = {
      code: 'USD',
      amount: 4567
    };

    expect(renderer.create(<CurrencyBalanceComponent noBalance={false} wallet={wallet}/>).toJSON()).toMatchSnapshot();
  });
});
