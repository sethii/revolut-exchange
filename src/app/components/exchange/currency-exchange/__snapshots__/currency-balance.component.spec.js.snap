// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency balance component renders currency balance 1`] = `
.glamor-1 {
  font-size: 12px;
  color: rgba(247, 0, 135, 1);
  font-family: Arial;
}

.glamor-1 span {
  font-size: 10px;
}

<p
  className="glamor-0 glamor-1"
>
  Balance: 
  <span>
    $
  </span>
  4567
</p>
`;

exports[`Currency balance component renders currency balance when not enough money available 1`] = `
.glamor-1 {
  font-size: 12px;
  color: #b1b1b1;
  font-family: Arial;
}

.glamor-1 span {
  font-size: 10px;
}

<p
  className="glamor-0 glamor-1"
>
  Balance: 
  <span>
    $
  </span>
  4567
</p>
`;
